### PR TITLE
Cover pkg/ipmi

### PR DESCRIPTION
**Pull request by ChriMarMe**
* Move structures to structure.go
* Move constants to constants.go
* Remove unused fdSet function
* Abstract syscall implementation behind interface
* Adapt implementation to new interface

### CODE REVIEW

File: pkg/ipmi/pkg_test.go [hunk 1]
```diff
@@ -4,8 +4,98 @@
```
Comment (walterchris) [inline pkg/ipmi/pkg_test.go:4]: Rename the file to ipmi_devapi_test.go ?

**Review by walterchris — APPROVED** [review]
LGTM - I would change the filename.

File: pkg/ipmi/ipmi_devapi.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package ipmi
+
+import (
+	"os"
+)
+
+type DevAPI interface {
```
Comment (jensdrenhaus) [inline pkg/ipmi/ipmi_devapi.go:11]: For me, it feels wrong to call that thing API, what about Handler? 
(To be honest, I have not found the perfect naming scheme in such cases, either)

File: pkg/ipmi/pkg_test.go (hunk 1 above)
Comment (jensdrenhaus) [reply pkg/ipmi/pkg_test.go:4]: +1

**Review by jensdrenhaus — CHANGES_REQUESTED** [review]
The introduction of DevAPI interface looks super. Many dependencies get decoupled. 
I think some methods of Dev struct (which you are not use during testing) can now still being tested on its own.

**Review by walterchris — APPROVED** [review]

**Comment by ChriMarMe** [comment]
[Open PR](https://github.com/u-root/u-root/pull/2182)

**Pull request closed without merging**